### PR TITLE
r-mass, r-cluster, r-class

### PR DESCRIPTION
**Pull request by johanneskoester**

### CODE REVIEW

**Comment by johanneskoester** [comment]
@bgruening I am now starting to migrate the R channel. As you suggested, I add you as maintainer to the feedstocks.

To whom it concerns: I have updated conda-build skeleton to reflect conda-forge needs, see this PR: conda/conda-build#1871.

File: recipes/r-class/bld.bat [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"%R%" CMD INSTALL --build .
+if errorlevel 1 exit 1
+
+@rem Add more build steps here, if they are necessary.
+
+@rem See
+@rem http://docs.continuum.io/conda/build.html
+@rem for a list of environment variables that are set during the build process.
```
Comment (ocefpaf) [inline recipes/r-class/bld.bat]: I usually remove these boilerplate comments. In fact I use the `script` line and drop the `bld.bat` and `build.sh` all together like what we do for Python.

**Comment by ocefpaf** [comment]
> To whom it concerns: I have updated conda-build skeleton to reflect conda-forge needs, see this PR: conda/conda-build#1871.

Thanks for that @johanneskoester but I guess that @mingwandroid's workflow is to automatic update all of `defaults` recipes by re-generating them. That is different from conda-forge, where we strive for a more "human readable recipe" and update manually with PRs changing the checksum and version number.

There is nothing wrong with either approach IMO but maybe we should try to get only the common (`conda-forge` and `defaults`) styles into `conda-skeleton` and apply the differences manually (or with a specialized tool).

PS: there is a GSoC student interested into extending `conda-skeleton` with `conda-forge` style btw.

**Comment by mingwandroid** [comment]
I regenerate all 250 r recipes at once and then scan the differences very carefully patching across any fixes.

If the skeleton is changed (as it already has, next time I do this I plan to revert those changes locally) then it will take me a lot longer.

I don't mind doing it once sometime if the format is improved significantly but I don't know when I'll have time for that, and I'm not sure any of the improvements so far have been significant.

In general. emitting more jinja2 into something that's generated anyway seems a bit pointless to me (though I am guilty of doing that too, and yes I did spot that this undoes some of that).

**Comment by ocefpaf** [comment]
> I regenerate all 250

@mingwandroid that is all we need to know to be careful with this :smile: 

The situation is not different for Python packages. We use `conda-skeleton` and then edit the recipe manually, and things worked out OK so far.

Ideally we should, slowly and without breaking your workflow, add things that make sense to both `defaults` and `conda-forge`. IMO having a second parser/extension to conda-skeleton that automate the `conda-forge` recipe style is not bad too.

**Comment by johanneskoester** [comment]
This is all fine for me, and indeed having the version as a variable is not necessary if this is done automatically. The only thing I really want to have in skeleton is the checksums.

File: recipes/r-class/bld.bat (hunk 1 above)
Comment (johanneskoester) [inline recipes/r-class/bld.bat]: We can remove the comments, but I don't want the build code in the meta.yaml. what I like about conda is the separation of build code and metadata.

**Comment by johanneskoester** [comment]
@conda-forge/core can this be merged?

**Comment by johanneskoester** [comment]
Thanks!

**Pull request merged**